pkg/client/client.go

```diff
@@ -0,0 +1,173 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/deltadefi-protocol/go-sdk/pkg/config"
+)
+
+type Client struct {
+	apiKey     string
+	NetworkId  uint8
+	Jwt        string
+	SigningKey string
+	HTTPClient *http.Client
+	BaseURL    string
+}
+
+func NewClient(apiKey string, network string, jwt string, signingKey string) *Client {
+	cfg := config.GetConfig()
+	var networkId uint8
+	var baseURL string
+
+	if network == "mainnet" {
+		networkId = uint8(1)
+		baseURL = "https://api-dev.deltadefi.io" // TODO: input production link once available
+	} else {
```

Comment:
better to throw error if network !== "mainnet" & "preprod" instead of default assigning dev link